code/modules/mob/living/ventcrawling.dm

```diff
@@ -73,6 +73,10 @@
 	else
 		var/datum/pipeline/vent_parent = ventcrawl_target.parents[1]
 		if(vent_parent && (vent_parent.members.len || vent_parent.other_atmos_machines))
+			var/datum/gas_mixture/internal_air = vent_parent.air
+			if(internal_air.temperature > get_body_temp_heat_damage_limit() || internal_air.temperature < get_body_temp_cold_damage_limit())
+				if(tgui_alert(src, "The air in this machine is hazardous to your health! (Temperature: [internal_air.temperature]!). Are you sure you want to enter?", "This will PROBABLY kill you!", list("Yes", "No")) != "Yes")
+					return
```

Comment:
You will require sanity checks after this alert, otherwise you can input stall to enter the vent from the opposite side of the map.